checkpoint.go

```diff
@@ -272,17 +271,5 @@ func Checkpoint(logger log.Logger, w *wal.WAL, m, n int, keep func(id uint64) bo
 	if err := closeAll(closers...); err != nil {
 		return stats, errors.Wrap(err, "close opened files")
 	}
-	if err := w.Truncate(n + 1); err != nil {
-		// If truncating fails, we'll just try again at the next checkpoint.
-		// Leftover segments will just be ignored in the future if there's a checkpoint
-		// that supersedes them.
-		level.Error(logger).Log("msg", "truncating segments failed", "err", err)
-	}
-	if err := DeleteCheckpoints(w.Dir(), n); err != nil {
-		// Leftover old checkpoints do not cause problems down the line beyond
-		// occupying disk space.
-		// They will just be ignored since a higher checkpoint exists.
-		level.Error(logger).Log("msg", "delete old checkpoints", "err", err)
-	}
```

Comment:
now that we have moved these should also update the Checkpoint func comment
remove - // Non-critical errors are logged and not returned.

Reply:
Removed. Also removed logger as an argument as it's no longer needed.

Reply:
Why did those get moved? This is an exported API with other tools depending on this to be in place.

Reply:
@krasi-georgiev I think we need to move it back to `Checkpoint`.